README.md

```diff
@@ -1,24 +1,256 @@
+            .build()
+
+```
+You should replace ```ABLY_API_KEY``` with your own. Please login into [Ably dashboard](https://ably.com/login) to retrieve
+ your own api key.
+ ```GAME_SCOPE``` is coroutine scope that you intend to run your AblyGame instance in. A Typical application scope
+ for that matter could be defined as
+ ```
+ val applicationScope = CoroutineScope(SupervisorJob() + Dispatchers.Main)
+ ```
+
+ #### Start a game
+
+```kotlinlang
+ablyGame.start {
+            when(it){
+                AblyGame.GameState.Started -> //TODO replace this with your action of game started
+                AblyGame.GameState.Stopped -> //TODO replace this with your action of game stopped
+            }
+        }
+```
+As you can see this method accepts a ```FlowCollector<GameState>``` for you to observe the changes on game state. You
+should make sure that the game is started before moving forward with other game functionalities
+
+#### Enter a game
+
+```
+someCoroutineScope.launch {
+            val enterResult = ablyGame.enter(gamePlayer)
+            if (enterResult.isSuccess) {
+               //enter successful
+            } else {
+               //enter failed
+            }
+        }
+```
+As you can see, you should invoke ```enter``` inside a coroutine scope.
+You must replace ```gamePlayer``` with your own instance of ```GamePlayer```
+
+#### Leave a game
+
+```kotlinlang
+someCoroutineScope.launch {
+            val leaveResult = MultiplayerGameApp.instance.ablyGame.leave(gamePlayer!!)
+            if (leaveResult.isSuccess) {
+                //leave successful
+            } else {
+               //leave failed
+            }
+        }
+```
+You must replace ```someCoroutineScope``` with scope your are launching the function.
+You must replace ```gamePlayer``` with your own instance of ```GamePlayer```
+
+#### Subscribing to game player updates in the game
+If you want to observe users entering / leaving a game, you can use the following example code
+
+```kotlinlang
+ablyGame.subscribeToGamePlayerUpdates {
+            when (it) {
+                is PresenceAction.Enter -> {
+                    Log.d(TAG, "PresenceAction.Enter ${it.player.id}")
+                }
+                is PresenceAction.Leave -> {
+                    Log.d(TAG, "PresenceAction.Leave ${it.player.id}")
+                }
+            }
+        }
+```
+
+#### Get list of all players
+You can get a list of all players in the game using the following code block
+```
+ coroutineScope.launch {
+            val players = ablyGame.allPlayers()
+       }
+```
+As ```allPlayers``` is a suspending function, you must call this from a coroutine scope
+
+#### Check if a player is in a game
+```
+coroutineScope.launch {
+            if (ablyGame.isInGame(gamePlayer)) {
+               //do something
+            }
+```
+
+### Game room related functions
+All game related functionality is inside ```GameRoomController```. You must get a handle of this to invoke
```

Comment:
Shouldn't it be `All game room related functionality`? :wink:

Reply:
Oops you are right